packages/cf.js/src/structs.ts

```diff
@@ -0,0 +1,38 @@
+import { ABIEncoding, Address } from "./simple-types";
+
+export enum NodeMessageType {
+  INSTALL = "install",
+  QUERY = "query",
+  ERROR = "error"
+}
+
+export enum QueryType {
+  GET_APP_INSTANCES = "getAppInstances"
+}
+
+export interface MessageDataQuery {
+  queryType: QueryType;
+  appInstances?: any[];
+}
+
+export interface MessageDataError {
+  message: string;
+  extra?: any;
```

Comment:
@mvanderh Should this be at least `object`?

Reply:
Maybe. `object` is kind of annoying though because you have to cast it before you can access itt like a JS object.

Reply:
You could do something like `{ [index: string]: any }`, maybe at least to suggest it has keys?

Reply:
Sure!